### ISSUE TITLE
`choose` crashes when in filter mode and you press `spacebar` twice

### ISSUE THREAD

**Issue by MikaelFangel** [charmbracelet/gum#580]
**Describe the bug**
When you use `choose`, press `/` to filter and then presses `spacebar` twice it crashed with an index out of bounds error.

**To Reproduce**
1. Run the command `gum choose --no-limit x y z`
2. Press `/` to enter filter
3. Press `spacebar` twice

**Expected behavior**
Filter shouldn't crashes from multiple spaces when in filter.

**Additional context**
Error message produced:
```
Caught panic:

runtime error: index out of range [0] with length 0

Restoring terminal...
```

**Comment by MikaelFangel** [comment]
Typing x repeatedly can also crash the program.

**Comment by meowgorithm** [comment]
Just a note that this is now available in [Gum v0.14.2](https://github.com/charmbracelet/gum/releases/tag/v0.14.2).
